fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -4711,6 +4716,7 @@ public void copyFrom(@Nonnull FDBRecordStore store) {
             this.metaDataProvider = store.metaDataProvider;
             this.context = store.context;
             this.subspaceProvider = store.subspaceProvider;
+            this.userVersionChecker = store.userVersionChecker;
```

Comment:
I think technically, there could be implementations of `UserVersionChecker` that store state, that could be confused about being reused on multiple store objects. That being said, re-reading the documentation for that interface, it seems like all the javadoc implies that it shouldn't be reused. 

It would also be good to follow up with more interesting coverage of `UserVersionChecker`, it is clear that we don't have good coverage if we drop the checker when converting to a builder and then back.